### ISSUE TITLE
Catch validation errors in save after a merge

### ISSUE THREAD

**Issue by ricmalta** [neumino/thinky#465]
When saving a merged document, I got this exception, but I can't handle it, its not trowed apparently:

The error I need to catch:

Unhandled rejection ReqlLogicError: Secondary keys must be a number, string, bool, pseudotype, or array (got {
	...
	 at Connection._processResponse (...api/node_modules/thinky/node_modules/rethinkdbdash/lib/connection.js:235:19)
    at Socket.<anonymous> (.../node_modules/thinky/node_modules/rethinkdbdash/lib/connection.js:156:14)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)

	
The code I've in my update:

```
Game.get(_id).run().then(function(game) {
      try {    
        game
        .merge(object)
        .save()
        .then(function(result){
          _results.push( result );
          return next(null, result);
        })
        .catch(next)
        .error(next);  
      } catch(e){
        log.error(e);
        return next();
      }
    })
    .catch(next)
    .error(next);
```
    
Notes:

The object that I'm testing is actually malformed and I need to catch the validation error to properly validate it.

I tried to to create new model object with the invalid object and throws no error too!
The error is happening in the save()

So basically I would like that it works like as the documentations says to: 	https://thinky.io/documentation/api/thinky/#errors
doing this approach:
https://thinky.io/documentation/api/document/#merge

**Comment by grantcarthew** [comment]
Try this:

```
Game.get(_id).run().then(function(game) {
        return game
        .merge(object)
        .validate()
        .save()
        .then(function(result){
          _results.push( result );
          return next(null, result);
        })
        .catch(next)
        .error(next);  
    })
    .catch(next)
    .error(next);
```
Note: Always return from inside a promise `return game`, and the `try/catch` is redundant.

**Comment by neumino** [comment]
@ricmalta -- can you provide the values and model definition that throw the error.

**Comment by ricmalta** [comment]
@grantcarthew 	It doesn't work:

TypeError: Cannot read property 'save' of undefined

**Comment by ricmalta** [comment]
@grantcarthew 
The
